### PR TITLE
Final touches and testing

### PR DESCRIPTION
**Pull request by MargaretTechworld**
In this Repository:

I have added a new component called Home.js, which is responsible for displaying a list of companies.
I have implemented a search functionality that enables users to search for companies based on their name.
I have also added a new page called CompanyDetails that displays detailed information about a selected company.
I have also added testing for all components.

- [Live Demo Link](https://graceful-beijinho-8c4721.netlify.app/)
- [Video Link](https://www.loom.com/share/beeec23d65f342b98d80dbe2bb38f3b6?sid=a23201a3-1a27-4d91-9521-74d0fa25af69)
- My design
image image

Thank you for your review.

### CODE REVIEW

File: README.md
Comment (fobadara) [inline README.md:1]: [REQUIRED] Kindy ensure it has an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/)

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @MargaretTechworld, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

REDO

## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt
✅ Good implementation of the principle of separating concerns.

## Required Changes ♻️
_Check the comments under the review._
I had to include some comments here because it could not be done inline

[REQUIRED] Kindly ensure that your design looks like the template in terms of composition and styling.
image
[Required] Kindly move the cards down and include some placeholder text or image just like the template
image

[Required] Your cards are more of squares than rectangles. kindly ensure that the width and height of your cards look like the template.

[Required] Your course details page doesn't look like the template. it should  have a top section with some text and image and the bottom section should be a sort of list with text on the left and right. Kindly ensure that the items are in rectangular boxes just like the design.
image

[REQUIRED] I noticed that I cant see some tests apart from snapshots. Kindly ensure that:
1. App has unit tests for several components of the app and some pure functions in Redux code.
2. App has integration tests using React Testing Library
And ensure that all tests are passing
   
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MargaretTechworld** [comment]
Hello, Sir. Thank you for your review. However, I want you to know that you are comparing my desktop version with the mobile version (original design). If you take a look at the mobile version, it looks responsive like the original design. I have made the necessary changes, and this is my design.
- My design
image image

File: src/test/Home.test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
```
Comment (Lucas-Masaba) [inline src/test/Home.test.js:1]: - [ ] You have failing tests from `Home.test.js` and `Navbar.test.js`. Please make sure all your tests are passing.
image

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @MargaretTechworld  👋 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost-there

 ## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Made changes as requested by the previous reviewer :+1: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/test/Home.test.js (hunk 1 above)
Comment (MargaretTechworld) [reply src/test/Home.test.js:1]: Hi, my tests are passing on my side. I'm worried about what the issue could be.
image

File: src/test/Navbar.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
```
Comment (AbayomiOlaoye) [inline src/test/Navbar.test.js:4]: - [ ] This is why your `Nav.test.js` is not passing. You have required/imported a package or project dependency that is not in your code therefore making it impossible to run perfectly. To fix this, I recommend that you run `npm i redux-mock-store` on your terminal or if you have, kindly push it to GitHub.

image

- Dependency missing in your `package.json` :point_down: 
image

**Review by AbayomiOlaoye — CHANGES_REQUESTED** [review]
## STATUS: CHANGE REQUIRED ♻️ 

Hi @MargaretTechworld :wave: , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

invalid

## Suggested changes
- [ ] _Check the comments under the review._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by OLIPLICHE — APPROVED** [review]
Hi @MargaretTechworld, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Fixed the baggs (✅)) 👏
- The test has passed 👏
- The App is working accordingly  👏 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**